nemoguardrails/library/content_safety/actions.py

```diff
@@ -0,0 +1,153 @@
+            "e.g. content safety check input $model=llama_guard"
+        )
+        raise ValueError(error_msg)
+
+    llm = llms.get(model_name, None)
+
+    if llm is None:
+        error_msg = (
+            f"Model {model_name} not found in the list of available models for content safety check. "
+            "Please provide a valid model name."
+        )
+        raise ValueError(error_msg)
+
+    task = f"content_safety_check_input $model={model_name}"
+
+    check_input_prompt = llm_task_manager.render_task_prompt(
+        task=task,
+        context={
+            "user_input": user_input,
+        },
+    )
+
+    stop = llm_task_manager.get_stop_tokens(task=task)
+    max_tokens = llm_task_manager.get_max_tokens(task=task)
+
+    llm_call_info_var.set(LLMCallInfo(task=task))
+
+    max_tokens = max_tokens or _MAX_TOKENS
+
+    with llm_params(llm, temperature=1e-20, max_tokens=max_tokens):
+        result = await llm_call(llm, check_input_prompt, stop=stop)
+
+    result = llm_task_manager.parse_task_output(task, output=result)
+
+    try:
+        is_safe, violated_policies = result
+    # in case the result is single value
+    except TypeError:
+        is_safe = result
+        violated_policies = []
+
+    return {"allowed": is_safe, "policy_violations": violated_policies}
+
+
+@action()
+async def content_safety_check_output(
+    llms: Dict[str, BaseLLM],
+    llm_task_manager: LLMTaskManager,
+    model_name: Optional[str] = None,
+    context: Optional[dict] = None,
+) -> dict:
+    _MAX_TOKENS = 3
+    user_input: str = ""
+    bot_response: str = ""
+
+    if context is not None:
+        user_input = context.get("user_message", "")
+        bot_response = context.get("bot_message", "")
+        model_name = context.get("model", None)
+
+    if model_name is None:
+        error_msg = (
+            "Model name is required for content safety check, "
+            "please provide it as an argument in the config.yml. "
+            "e.g. flow content safety (model_name='llama_guard')"
+        )
+        raise ValueError(error_msg)
+
+    llm = llms.get(model_name, None)
+
+    if llm is None:
+        error_msg = (
+            f"Model {model_name} not found in the list of available models for content safety check. "
+            "Please provide a valid model name."
+        )
+        raise ValueError(error_msg)
+
+    task = f"content_safety_check_output $model={model_name}"
+
+    check_output_prompt = llm_task_manager.render_task_prompt(
+        task=task,
+        context={
+            "user_input": user_input,
+            "bot_response": bot_response,
+        },
+    )
+    stop = llm_task_manager.get_stop_tokens(task=task)
+    max_tokens = llm_task_manager.get_max_tokens(task=task)
+
+    max_tokens = max_tokens or _MAX_TOKENS
+
+    llm_call_info_var.set(LLMCallInfo(task=task))
+
+    with llm_params(llm, temperature=1e-20, max_tokens=max_tokens):
```

Comment:
gemmashield or ChatNVIDIA has `gt = 0.0` constraint on `max_tokens` field and not `ge`.

Reply:
Ack.